test/build/config/type/function-with-env/function-with-env.test.js

```diff
@@ -142,11 +142,35 @@ describe("function configuration", () => {
 
     expect(exitCode).toBe(0);
     expect(stderr).toBeFalsy();
-    expect(stdout).toBeTruthy();
-    // Should generate the appropriate files
-    expect(existsSync(resolve(__dirname, "./dist/equal-at-the-end.js"))).toBeTruthy();
+    // should log foo: undefined
+    expect(stdout).toContain("foo: undefined");
+  });
+
+  it('Supports env variable with "=$NON_EXISTENT_VAR" at the end', async () => {
+    const { exitCode, stderr, stdout } = await run(__dirname, ["--env", `foo=$NON_EXISTENT_VAR`], {
+      shell: true,
+    });
+
+    expect(exitCode).toBe(0);
+    expect(stderr).toBeFalsy();
+    // should log foo: undefined
+    expect(stdout).toContain("foo: undefined");
   });
 
+  // macOS/Linux specific syntax
+  if (!isWindows) {
+    it('Supports env variable with "foo=undefined" at the end', async () => {
+      const { exitCode, stderr, stdout } = await run(__dirname, ["--env", `foo=undefined`]);
+
+      expect(exitCode).toBe(0);
+      expect(stderr).toBeFalsy();
+      // should log foo: 'undefined'
+      expect(stdout).toContain("foo: 'undefined'");
+      // Should generate the appropriate files
+      expect(existsSync(resolve(__dirname, "./dist/undefined-foo.js"))).toBeTruthy();
+    });
+  }
```

Comment:
And lets add
```
webpack --env TEST=""
```
```
webpack --env TEST=\\\"\\\"
```

To avoid future regressions

Reply:
@alexander-akait Done ✅